### ISSUE TITLE
Bug: Widget width too small on KDE Plasma Version: 5.26.0

### ISSUE THREAD

**Issue by Everest10** [blackadderkate/weather-widget-2#116]
Operating System: Fedora Linux 37
KDE Plasma Version: 5.26.0
KDE Frameworks Version: 5.99.0
Qt Version: 5.15.6
Graphics Platform: X11

Bug: Widget width too small.

![image](https://user-images.githubusercontent.com/7228207/196057005-a241550d-64f4-442d-99d2-c39328aa4680.png)

**Comment by Nowruz** [comment]
Same issue here.

Operating System: Arch Linux
KDE Plasma Version: 5.26.0
KDE Frameworks Version: 5.99.0
Qt Version: 5.15.6
Kernel Version: 6.0.2-arch1-1 (64-bit)
Graphics Platform: X11

**Comment by blackadderkate** [comment]
Duplicate of issue #109 .
Fixed in [Version 2.3.3](https://github.com/blackadderkate/weather-widget-2/releases/tag/v2.3.3) (now also available from the KDE Store)

**Comment by fkru** [comment]
> Duplicate of issue #109 . Fixed in [Version 2.3.3](https://github.com/blackadderkate/weather-widget-2/releases/tag/v2.3.3) (now also available from the KDE Store)

Just installed 2.3.3, still having this issue. On top of that, KDE Store's version deleted all my local settings.

**Comment by fkru** [comment]
> > Duplicate of issue #109 . Fixed in [Version 2.3.3](https://github.com/blackadderkate/weather-widget-2/releases/tag/v2.3.3) (now also available from the KDE Store)
> 
> Just installed 2.3.3, still having this issue. On top of that, KDE Store's version deleted all my local settings.

Strange, after renaming /usr/share/plasma/plasmoids/org.kde.plasma.weather/contents/ui/FullRepresentaion.qml it works as expected and the issue is gone. What do I miss?

**Comment by blackadderkate** [comment]
> > Duplicate of issue #109 . Fixed in [Version 2.3.3](https://github.com/blackadderkate/weather-widget-2/releases/tag/v2.3.3) (now also available from the KDE Store)
> 
> Just installed 2.3.3, still having this issue. On top of that, KDE Store's version deleted all my local settings.

Regarding the deleting of your settings:
There's virtually no documentation on how to publish things on the KDE Store (pling.com) - I think maybe I did something wrong in the publishing process and Discover is treating it as a new download rather than an upgrade?

**Comment by Everest10** [comment]
Uninstalled the old version and deleted the /usr/share/plasma/plasmoids/org.kde.plasma.weather till same problem in 2.3.3

**Comment by fkru** [comment]
> > > Duplicate of issue #109 . Fixed in [Version 2.3.3](https://github.com/blackadderkate/weather-widget-2/releases/tag/v2.3.3) (now also available from the KDE Store)
> > 
> > 
> > Just installed 2.3.3, still having this issue. On top of that, KDE Store's version deleted all my local settings.
> 
> Regarding the deleting of your settings: There's virtually no documentation on how to publish things on the KDE Store (pling.com) - I think maybe I did something wrong in the publishing process and Discover is treating it as a new download rather than an upgrade?

Maybe, unfortunately, I don't have insight into KDE Store. By the way, where are the local settings such as location etc. stored?

**Comment by kppr0** [comment]
Try uninstall, delete: /home/your_account/.local/share/plasma/plasmoids/org.kde.weatherWidget-2, restart and install new widget version.

**Comment by blackadderkate** [comment]
> Uninstalled the old version and deleted the /usr/share/plasma/plasmoids/org.kde.plasma.weather still same problem in 2.3.3

@Everest10: The location depends on how the widget was installed...

If installed by "Get Widget" it will install your home folder in the following location: `$HOME/.local/share/plasma/plasmoids/org.kde.weatherWidget-2`

If installed using AUR/pacman it will install for all users in `/usr/share/plasma/plasmoids/org.kde.weatherWidget-2`

Make sure both those folders are gone, [download the plasmoid again](https://github.com/blackadderkate/weather-widget-2/releases/tag/v2.3.3) and install using `plasmapkg -i` ...

**Comment by blackadderkate** [comment]
> By the way, where are the local settings such as location etc. stored?

@fkru they are stored in $HOME/config/plasma-org.kde.plasma.desktop-appletsrc
Look for a section starting something like this (although the [2] and [38] will vary according to how you have configured your desktop):

```
[Containments][2][Applets][38]
immutability=1
plugin=org.kde.weatherWidget-2
```
Locations settings are in this section
...
[Containments][2][Applets][38][Configuration][General]
...

**Comment by ZWindL** [comment]
It appears too big on my laptop now.
Distro: Arch Linux
KDE Plasma Version: 5.26.1
KDE Frameworks Version: 5.99.0
QT Version: 5.15.6
Graphics Platform: X11
UI Scale: 250%
Resolution: 3840x2160 
DPI: 210

This package is installed from AUR, and I tried to remove widget -> uninstall package -> install again -> add widget, still that big.

It fills the whole screen, I 
![Screenshot_20221021_161517](https://user-images.githubusercontent.com/5116996/197147551-e1c9a115-8322-404d-8a32-21cee1717be5.png)

**Comment by Everest10** [comment]
It's ok right now. Fedora 37, Plasma 5.26.1

**Comment by blackadderkate** [comment]
@ZWindL OK, I'll need to try different UI Scale and DPI settings on my development Virtual Machine, to see if I can figure out what is going on.

**Comment by Grief** [comment]
Same problem, widget appears too big with overlapped labels
![image](https://user-images.githubusercontent.com/7239315/201649720-8ff7422a-2a18-4f42-9e01-7fd34501e7a4.png)

**Comment by blackadderkate** [comment]
@Grief  On my development machine, devicePixelRatio is 1 regardless of what I set DPI or the Global Scale to. Presumably this setting only changes if you have an Ultra HD monitor, which I don't have.

I'll remove the multipliers as you suggested...

Longer term, I'd like to implement some kind of scaling setting, so that the Meteogram can be made bigger on very large screens... I'll raise a new issue for that now...

**Comment by Grief** [comment]
You are correct, devicePixelRatio is for monitor's dpi (my screen is 4K), however there are a lot of tricks with it: https://doc.qt.io/qt-5/highdpi.html

**Comment by ArchangeGabriel** [comment]
The DPI change is borking the display on HiDPI monitors indeed: https://github.com/blackadderkate/weather-widget-2/issues/127#issuecomment-1321099183

**Comment by phyrz91** [comment]
Hi, same problem:
On X11 there is no problem.
[url=https://tinypic.host/i/Kqn5g][img]https://tinypic.host/images/2023/01/23/wwidget2.md.png[/img][/url]

Version: 2.3.4.3

**Comment by Grimmicks** [comment]
> [@blackadderkate](https://github.com/blackadderkate) I've removed multiplications by `devicePixelRatio` all across the code and it started to look adequate. Could you please add an option, like `☑ Ignore system DPI` into the appearance settings tab which would make those multiplications conditional?

Just wanted to add I had this same issue and this fixed it for me. Removed `devicePixelRatio` multipliers from `imageWidth` and `imageHeight` in the FullRepresentation.qml file.

Debian 12
Plasma 5.27 Wayland
